core/modules/replica/Configuration.h

```diff
@@ -43,364 +44,365 @@
 
-    void dumpIntoLogger() const final;
+    /**
+     * The static factory method will create an object and initialize its transient
+     * state from the given JSON object. The parameters will amend the default state
+     * of the object.
+     * @note Configuration objcts created by this method won't have any persistent
+     *   backend should any changes to the transient state be made.
+     * @param obj The input configuration parameters. The object is optional.
+     *   If it's not given (the default value), or if it's empty then no changes
+     *   will be made to the transient state.
+     * @throw std::runtime_error If the input configuration is not consistent
+     *   with expectations of the application.
+     */
+    static Ptr load(nlohmann::json const& obj=nlohmann::json::object());
 
-    // ---------------------------------------------------------------
-    // -- (global) parameters of the Qserv worker database services --
-    // ---------------------------------------------------------------
+    /**
+     * The static factory method will create a new object and initialize its content
+     * from the following source:
+     * @code
+     *   mysql://[user][:password]@[host][:port][/database]
+     * @code
+     * @note Configuration objects initialized from MySQL would rely opn MySQL as
+     *   the persistent backend for any requests to update the state of the transient
+     *   parameters. A connection object to the MySQL service will be initialized
+     *   at the corresponding data member of the class.
+     * @param configUrl The configuration source.
+     * @param autoMigrateSchema The optional flag that if 'true' would result in an attempt
+     *   to automatically migrate the database schema up to the level of the current
+     *   configuration's should the persistent stores schema versin be lower than
+     *   the expected one. Note that no schema downgrade is supported in the current
+     *   implementation. The flag is supposed to be set to 'true' when instantiating
+     *   the configuration object in the Master Replication Controller.
+     * @throw std::invalid_argument If the URL has unsupported scheme or it
+     *   couldn't be parsed.                          
+     * @throw std::runtime_error If the input configuration is not consistent
+     *   with expectations of the application.
+     */
+    static Ptr load(std::string const& configUrl, bool autoMigrateSchema=false);
 
-    /// @return the database password
+    /// @return The database password for accessing Qserv czar's database.
     static std::string const& qservMasterDatabasePassword() { return _qservMasterDatabasePassword; }
 
     /**
-     * @param newPassword The new password to be set.
-     * @return the previous value of the password
+     * @param newPassword The new password to be set for accessing Qserv czar's database.
+     * @return The previous value of the password.
```

Comment:
Just a coding pattern for the "setter" methods. It could be used for tracking changes in the configuration and logging these events into log streams. Perhaps it's not really needed here.